common/src/node/ethereum/utils.rs

```diff
@@ -235,33 +239,49 @@ pub async fn get_contract_balance(
     ContractCall::from(call).call().await
 }
 
+fn create_localwallet_client(
+    polling_interval_ms: u64,
+    key: Arc<SecretKey>,
+    chain_id: u64,
+    client: Provider<Http>,
+) -> Result<SignerMiddleware<Provider<Http>, Wallet<SigningKey>>, EthError> {
+    let provider = client.interval(Duration::from_millis(polling_interval_ms));
+    let wallet = LocalWallet::from(key.get_signing_key()).with_chain_id(chain_id);
+    let client = SignerMiddleware::new(provider, wallet);
+    Ok(client)
+}
 /// given the contract approval details, it'll construct, sign and broadcast a
 /// corresponding approval transaction.
 /// If successful, it returns the transaction receipt.
```

Comment:
I guess this description isn't fully correct -- it'll only sign if the secret key is provided?
and similarly, the name `broadcast` isn't very intuitive, because it won't always broadcast?


The Ok return type may be a little bit confusing -- it seems it won't ever return `(None, None)` or `(Some(receipt), Some(tx))` ?

I think this was based on @damoncro 's previous feedback to combine the methods: https://github.com/crypto-com/defi-wallet-core-rs/pull/620#pullrequestreview-1144994457 not sure if he meant it like that.

Anyway, it's good if the common functionality is combined/not duplicated, but the public methods should have meaningful names, types etc.

Some options:
- move this combined stuff into private functions, and have two public functions that have more meaningful names and return types
- or rename the public function, and return some enum that only allows `EthersTransactionReceipt` or `TypedTransaction`

Reply:
ok, i'll refactor more for broadcast

Reply:
Keep broadcast as it is (it needs private key), but call build_tx../construct_tx (accpets Option key) inside broadcast.

Reply:
i'll restore to  original broadcast_, and change build_.. to construct_...
make it to return EthersTransactionReceipt or TypedTransaction